docsrc/providers.rst

```diff
@@ -467,3 +468,48 @@ If your app uses storage passwords (like the Qualys app), the distribution stanz
 
 
 Uses client :ref:`Splunk <client_splunk>`.
+
+Splunk
+======
+
+Provides rotation support for Splunk Http Event Collector (HEC) tokens. Distribution is not supported by this provider.
+
+Like the Qualys provider, the spec requires you to specify a `rotatewith` value, which is where Keydra can
+find creds with which to access Splunk and make the change. 
+
+.. code-block:: yaml
+
+   key: hec1
+   description: Splunk HEC Rotation Example
+   custodians: your_team
+   provider: splunk
+   rotate: nightly
+   config:
+      host: your.splunkhostname.com
+      rotatewith:
+         key: keydra/splunk/admin
+         provider: secretsmanager
+   distribute:
+   -
+      key: keydra/splunk/hec1
+      provider: secretsmanager
+      source: secret
+      envs:
+         - prod
+
+The Secrets Manager entry format is as follows:
+
+.. code-block:: yaml
+
+   {
+   "provider": "splunk",
+   "key": "HEC Input Name",
```

Comment:
I'm a little confused by the `key` here -- is this the input name that we expect to match on `src/keydra/clients/splunk.py:230`? If so, should `src/keydra/providers/splunk_hec.py:73` pass `self._credentials['key']`  (which is part of the secret blob) rather than `secret['key']` (which is the config spec)? Otherwise this `key` field doesn't appear to be used, like `provider`.

I'd consider renaming some of these. E.g. within the secret we could call it something like `hecInputName` instead of `key` to make it unambiguous; and `hecToken` instead of `secret`.

Could also rename the vars within the provider class (`secret` -> `spec`), but that's a smaller concern.

Reply:
Fair enough. The `key` from the spec is the one that is used, but I guess I wanted it saved into the SM secret so there's a way to see which input that token relates to from there.

I'll change the SM secret key names to make it more intuitive, and remove the provider key - this is not needed at all.

Reply:
> The key from the spec is the one that is used

Would be good to make it clear in the docs somewhere :) Right now it looks like we only refer to "input name" within the docs when describing the contents of the secret. 

Although to me it would make more sense to separate the two -- the spec `key` is used to locate the secret in SM, and I don't believe we use it for any other purposes in other providers. I think drawing that boundary between providers and core Keydra could avoid some confusion.